logic/createGoogleCalendar.js

```diff
@@ -0,0 +1,273 @@
+  //calendarInfo.whichGoogleAccountInteger = 1;
+  //  }
+  // const readline = require("readline");
+  const { google } = require("googleapis"); // just downloads the 'google' part out of this large googleaplis (destructuring)
+  // const SCOPES = ["https://www.googleapis.com/auth/calendar"];
+
+  // Assign the proper Google Auth account based on 'whichGoogleAccountInteger'
+  const token = process.env.TOKEN1 || "No Token Found in env file";
+  const credentials = process.env.CREDENTIALS1 || "No Token Found in env file";
+
+  // if (calendarInfo.whichGoogleAccountInteger === 2) {
+  //   token = process.env.TOKEN2 || "No Token Found in env file";
+  //   credentials = process.env.CREDENTIALS2 || "No Token Found in env file";
+  // } else if (calendarInfo.whichGoogleAccountInteger === 3) {
+  //   token = process.env.TOKEN3 || "No Token Found in env file";
+  //   credentials = process.env.CREDENTIALS3 || "No Token Found in env file";
+  // } else {
+  //   token = process.env.TOKEN1 || "No Token Found in env file";
+  //   credentials = process.env.CREDENTIALS1 || "No Token Found in env file";
+  // }
+
+  function authorize(credentials, callback) {
+    // eslint-disable-next-line camelcase
+    const { client_secret, client_id, redirect_uris } = credentials.installed;
+    const oAuth2Client = new google.auth.OAuth2(
+      client_id,
+      client_secret,
+      // eslint-disable-next-line camelcase
+      redirect_uris[0]
+    );
+    oAuth2Client.setCredentials(JSON.parse(token));
+    callback(oAuth2Client);
+  }
+  /* eslint-enable */
+
+  authorize(JSON.parse(credentials), insertCalendar);
+  //****************************************** Custom Functions After Authorization completed above
+
+  function insertCalendar(auth) {
+    logthis(
+      "Trying to Create/Insert New Calendar, Named = " + eventData.raceName
+    );
+
+    //here we go!
+    calendar = google.calendar({ version: "v3", auth });
+    calendar.calendars.insert(
+      {
+        resource: { summary: eventData.raceName }
+      },
+      function(err, res) {
+        if (err) {
+          respond(null);
+          return console.log("The API returned an error: " + err);
+        }
+        // var calenderInfo = calendarInfo;
+
+        // calenderInfo.summary= res.data.summary;
+        console.log(
+          "\n\nSuccessfully created Calendar Named : " +
+            res.data.summary +
+            "\n  with id # " +
+            res.data.id +
+            "\n and etag: " +
+            res.data.etag
+        );
+        calendarInfo.calendarId = res.data.id;
+        //        respond(calendarInfo.calendarId);
+        calendarInfo.etag = res.data.etag;
+
+        shareCalendar(auth, calendarInfo);
+        return calendarInfo;
+      }
+    );
+  }
+
+  function shareCalendar(auth, calendarInfo) {
+    if (gmailAddress) {
+      console.log("Now trying to share Calendar with: " + gmailAddress);
+    } else {
+      gmailAddress = "perrywilliams@800appliance.com";
```

Comment:
You should probably change this to joglog@gmail.com or something unless you love spam emails